plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -425,6 +427,16 @@ public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTable
             }
             throw e;
         }
+        ProtocolEntry protocolEntry = metastore.getProtocol(session, metastore.getSnapshot(dataTableName, session));
```

Comment:
reuse `tableSnapshot` obtained above.

This will help reverting the changes from `io.trino.plugin.deltalake.TestDeltaLakeFileOperations`